File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res = curSum = float("-inf")
        left = 0

        for right in range(len(nums)):
            if curSum < 1:
                left = right
                curSum = 0

            curSum += nums[right]
            res = max(res, curSum)

        return res

File: 0053-maximum-subarray/0053-maximum-subarray.go
func maxSubArray(nums []int) int {
    res , curSum := math.MinInt16, 0

    for i := 0; i < len(nums); i++ {
        if curSum < 0 {
            curSum = 0
        }

        curSum += nums[i]
        if curSum > res {
            res = curSum
        }
    }   

    return res
}